### PR TITLE
DOC: a.size and np.prod(a.shape) are not equivalent

### LINKED ISSUE #10747: a.size and np.prod(a.shape) are not equivalent
The documentation of `ndarray.size` claims that `a.size` is equivalent to `np.prod(a.shape)`. While they do return the same value, they return different types and therefore are not equivalent.

```python
a = np.zeros((2, 2))

type(a.size)
# returns <class 'int'>

type(np.prod(a.shape))
# returns <class 'numpy.int64'>
```

Continuing the calculations using `np.prod(a.shape)` can therefore lead to integer overflows.

I guess `np.prod(a.shape)` cannot be changed to return a `<class 'int'>` instance, and `a.size` should certainly not be changed to return a `<class 'numpy.int64'>` instance, so probably the only solution is to update the documentation to reflect that these two are not fully equivalent.

I'd be happy to make pull-request to update the documentation accordingly.

### PR DESCRIPTION
**Pull request by jonasrauber**
fixes #10747

### CODE REVIEW

File: numpy/add_newdocs.py [hunk 1]
```diff
@@ -3060,8 +3060,10 @@ def luf(lamdaexpr, *args, **kwargs):
     """
     Number of elements in the array.
 
-    Equivalent to ``np.prod(a.shape)``, i.e., the product of the array's
-    dimensions.
+    Equal to ``np.prod(a.shape)``, i.e., the product of the array's
+    dimensions, except that ``size`` returns an instance of ``int``
+    whereas ``np.prod(a.shape)`` returns an instance of ``np.int_``
+    (e.g. ``np.int64``).
```
Comment (jaimefrio) [inline numpy/add_newdocs.py]: I think this obscures the information being provided a little. I'd suggest keeping the original sentence unchanged, perhaps with `s/Equivalent/Equal/`, and then add the type observation as a separate paragraph, or even in a Notes section. For most users the distinction is not really relevant, so I don't think it deserves equal footing with the much more useful "here's how this value is calculated."

I would prefer something along the lines of this:

```
Notes
-----
`a.size` returns a standard arbitrary precision Python integer. This 
may not be the case with other methods of obtaining the same value
(like the suggested ``np.prod(a.shape)``, which returns an instance
of ``np.int_``), and may be relevant if the value is used further in
calculations that may overflow a fixed size integer type.
```

**Review by jaimefrio — CHANGES_REQUESTED** [review]

**Comment by jonasrauber** [comment]
Integrated the changes proposed by @jaimefrio

File: numpy/add_newdocs.py [hunk 2]
```diff
@@ -3070,6 +3070,14 @@ def luf(lamdaexpr, *args, **kwargs):
...
     >>> np.prod(x.shape)
     30
+    
+    Notes
+    -----
+    `a.size` returns a standard arbitrary precision Python integer. This 
+    may not be the case with other methods of obtaining the same value
+    (like the suggested ``np.prod(a.shape)``, which returns an instance
+    of ``np.int_``), and may be relevant if the value is used further in
+    calculations that may overflow a fixed size integer type.
```
Comment (jaimefrio) [inline numpy/add_newdocs.py]: `Notes` seems to always come before `Examples` in all other docstrings. Could you please swap the order to keep consistency?

**Review by jaimefrio — COMMENTED** [review]
Thanks for the quick fix, we're almost there!
One last nitpick and I'll hit the green button.

File: numpy/add_newdocs.py (hunk 2 above)
Comment (jonasrauber) [reply numpy/add_newdocs.py]: Fixed

**Review by jaimefrio — APPROVED** [review]

**Comment by jaimefrio** [comment]
Thanks Jonas, in it goes!

**Pull request merged**